src/promise/Promise.php

```diff
@@ -52,4 +53,54 @@ public function onCompletion(\Closure $onSuccess, \Closure $onFailure) : void{
 	public function isResolved() : bool{
 		return $this->shared->resolved;
 	}
+
+	/**
+	 * Returns a promise that will resolve only once all the Promises in
+	 * `$promises` have resolved. The resolution value of the returned promise
+	 * will be an array containing the resolution values of each Promises in
+	 * `$promises` indexed by the respective Promises' array keys.
+	 *
+	 * @template TPromiseValue
+	 * @template TKey of array-key
+	 * @phpstan-param array<TKey, Promise<TPromiseValue>> $promises
+	 *
+	 * @phpstan-return Promise<array<TKey, TPromiseValue>>
+	 */
+	public static function all(array $promises) : Promise {
+		/** @phpstan-var PromiseResolver<array<TKey, TPromiseValue>> $resolver */
+		$resolver = new PromiseResolver();
+		$values = [];
+		$toResolve = count($promises);
+		$continue = true;
+
+		foreach($promises as $key => $promise){
+			$promise->onCompletion(
+				function(mixed $value) use ($resolver, $key, &$toResolve, &$continue, &$values) : void{
+					$values[$key] = $value;
+
+					$toResolve--;
+					if($toResolve === 0 && $continue){
```

Comment:
It should be impossible for `$toResolve` to reach `0` if any of the promises were rejected, right? So it seems to me like the right side of this condition is actually unnecessary anyway.

Reply:
I thought PHPStan was bugged in its handling of this, but I'm starting to wonder if it actually played a galaxy brain move by finding this

Reply:
No, the individual promises could still resolve correctly.
I'm saying the `$continue` check is redundant, since `$toResolve` cannot reach `0` in any condition where `$continue` becomes `false`.

Reply:
You're right